aiida/common/timezone.py

```diff
@@ -8,132 +8,76 @@
 
-def localtime(value, timezone=None):
-    """Converts an aware datetime.datetime to local time.
-
-    Local time is defined by the current time zone, unless another time zone is specified.
+    :param name: The timezone name. Should correspond to a known name in the Olsen database.
+        https://en.wikipedia.org/wiki/Tz_database
+    :returns: The corresponding timezone object.
+    :raises ValueError: if the timezone name is unknown.
     """
-    if timezone is None:
-        timezone = get_current_timezone()
-
-    # If `value` is naive, astimezone() will raise a ValueError, so we don't need to perform a redundant check.
-    value = value.astimezone(timezone)
-    if hasattr(timezone, 'normalize'):
-        # This method is available for pytz time zones.
-        value = timezone.normalize(value)
-
-    return value
+    try:
+        return pytz.timezone(name)
+    except pytz.exceptions.UnknownTimeZoneError as exception:
+        raise ValueError(f'unknown timezone: {name}') from exception
 
 
-def delta(from_time, to_time=None):
+def delta(from_time: datetime, to_time: datetime = None) -> timedelta:
     """Return the datetime object representing the different between two datetime objects.
 
-    :param from_time: starting datetime object
-    :param to_time: end datetime object
-    :return: the delta datetime object
+    :param from_time: The starting datetime object.
+    :param to_time: The end datetime object. If not specified :func:`aiida.common.timezone.now` is used.
+    :return: The delta datetime object.
     """
-    if to_time is None:
-        to_time = now()
-
-    try:
-        from_time_aware = make_aware(from_time)
-    except ValueError:
-        from_time_aware = from_time
-    try:
-        to_time_aware = make_aware(to_time)
-    except ValueError:
-        to_time_aware = to_time
-
-    return to_time_aware - from_time_aware
+    return make_aware(from_time) - make_aware(to_time or now())
 
 
-def datetime_to_isoformat(value):
+def datetime_to_isoformat(value: datetime) -> str:
     """Convert a datetime object to string representations in ISO format.
 
-    :param value: a datetime object
+    :param value: A datetime object
     """
-    if value is None:
-        return None
-    return value.isoformat()
+    return value.isoformat() if value is not None else value
 
 
-def isoformat_to_datetime(value):
+def isoformat_to_datetime(value: str) -> datetime:
     """Convert string representation of a datetime in ISO format to a datetime object.
 
-    :param value: a ISO format string representation of a datetime object
+    :param value: An ISO format string representation of a datetime object.
+    :raises ValueError: if the provided string does not have a valid ISO format.
     """
-    if value is None:
-        return None
-    return parser.parse(value)
+    return datetime.fromisoformat(value) if value is not None else value
```

Comment:
Idem the inverse method, we can discuss there and just leave this comment as reminder to also consider it here.